### PR TITLE
Fix esql build script configuration cache incompatibilities

### PR DESCRIPTION
**Pull request by breskeby**

### CODE REVIEW

File: x-pack/plugin/esql/src/main/generated/org/elasticsearch/xpack/esql/expression/function/scalar/ip/IpPrefixEvaluator.java [hunk 1]
```diff
@@ -42,12 +42,12 @@ public IpPrefixEvaluator(Source source, EvalOperator.ExpressionEvaluator ip,
       EvalOperator.ExpressionEvaluator prefixLengthV4,
       EvalOperator.ExpressionEvaluator prefixLengthV6, BytesRef scratch,
       DriverContext driverContext) {
-    this.warnings = new Warnings(source);
     this.ip = ip;
     this.prefixLengthV4 = prefixLengthV4;
     this.prefixLengthV6 = prefixLengthV6;
     this.scratch = scratch;
     this.driverContext = driverContext;
+    this.warnings = Warnings.createWarnings(driverContext.warningsMode(), source);
```
Comment (mark-vieira) [inline x-pack/plugin/esql/src/main/generated/org/elasticsearch/xpack/esql/expression/function/scalar/ip/IpPrefixEvaluator.java:50]: Is this change related somehow?

Comment (breskeby) [reply x-pack/plugin/esql/src/main/generated/org/elasticsearch/xpack/esql/expression/function/scalar/ip/IpPrefixEvaluator.java]: that was a timing issue with my rebase. fixed

File: x-pack/plugin/esql/build.gradle [hunk 2]
```diff
@@ -47,24 +47,34 @@ tasks.named("compileJava").configure {
...
 }
 
+interface Injected {
+  @Inject FileSystemOperations getFs()
+}
+
 tasks.named("test").configure {
   if (BuildParams.isCi() == false) {
     systemProperty 'generateDocs', true
+    def injected = project.objects.newInstance(Injected)
```
Comment (mark-vieira) [inline x-pack/plugin/esql/build.gradle:57]: Instead of all this can we just grab a handle `project.getFileOperations()`?

**Review by mark-vieira — APPROVED** [review]
